pkg/web/apps/apps_psql_repository.go

```diff
@@ -128,5 +130,154 @@ func (a *appsPostgreSQLRepository) GetAppByID(ID string) (*models.App, error) {
 	}
 
 	return &app, nil
+}
+
+// GetDeviceByDeviceID gets a device by its Device ID
+func (a *appsPostgreSQLRepository) GetDeviceByDeviceID(deviceID string) (*models.Device, error) {
+	query := `
+	SELECT id,version_id,app_id,device_id,device_type,device_version
+	FROM device
+	WHERE device_id = $1;`
+
+	row := a.db.QueryRow(query, deviceID)
+
+	var device models.Device
+	if err := row.Scan(&device.ID, &device.VersionID, &device.AppID, &device.DeviceID, &device.DeviceType, &device.DeviceVersion); err != nil {
```

Comment:
There are some differences in the formatting of our error handling between different functions.

This function includes the row.Scan with the err conditional and is using a switch statement for the sql.ErrNoRows error check. The GetDeviceByDeviceIDAndAppID function has the QueryRow, Scan and err conditional on the same line and is not using a switch statement for the sql.ErrNoRows check. The other functions are using some version of the above aswell.

I don't have a preference on which one is chosen, the main goal is consistency. Even though it is a small thing.

Reply:
I agree. I will update the ones involved in this PR. There are probably inconsistencies elsewhere which I will not touch here.